### PR TITLE
Rename init time shmem allocation function

### PR DESCRIPTION
**Pull request by gcongiu**
## Pull Request Description

We rename `MPIDU_shm_seg_*` to `MPIDU_Init_shm_*` to remark that these APIs are only used to `alloc/free` shared memory segments at MPI init time. By doing so we make the `MPIDU_shm` namespace available to implement runtime allocation function that uses a similar mechanism as for pt2pt also for RMA shared memory windows.

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Remove xfail from the test suite when fixing a test
* [ ] Commits are self-contained and do not do two things at once
* [ ] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [ ] Passes whitespace checkers
* [ ] Passes warning tests
* [ ] Passes all tests
* [ ] Add comments such that someone without knowledge of the code could understand
* [ ] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

**Comment by gcongiu** [comment]
test:jenkins/ch3/tcp
test:jenkins/ch4/ofi

**Comment by gcongiu** [comment]
@minsii what do you think about moving the shared memory allocation functionalities in RMA to `MPIDU_shm_alloc` in `src/mpid/common/shm`? I started looking at how to integrate memory binding support for shm_win and I found it difficult to proceed without doing some code restructuring first. This PR is should be the first step in that direction.

**Comment by gcongiu** [comment]
test:jenkins/ch3/tcp

**Comment by minsii** [comment]
> integrate

Yes, the plan sounds great to me. Just be careful that `win_allocate` and `win_allocate_shared` both use the shared window allocation. Both try two allocation methods (if 1 fails it will try 2)
1. Global symm heap with shm segment on each node (see `MPIDIU_get_shm_symheap`): every process maps to the same start address.
2. Normal shm segment (see `MPIDIU_allocate_shm_segment`): the start address on each process might be different.

If both the above methods fail, `win_allocate` will allocate non-shm memory, but `win_allocate_shared` will fail.

Just let me know if you have any question when changing the code.

**Comment by gcongiu** [comment]
CH3 test results: https://jenkins-pmrs.cels.anl.gov/job/mpich-review-ch3-tcp/224/

**Comment by gcongiu** [comment]
Running CH4 too:

test:jenkins/ch4/ofi

**Comment by gcongiu** [comment]
test:jenkins/ch4/ofi

**Comment by gcongiu** [comment]
@minsii can you look at this one for now? It is very short so it should not take long.

File: src/mpid/ch3/channels/nemesis/include/mpid_nem_defs.h [hunk 1]
```diff
@@ -10,7 +10,7 @@
 #include "mpid_nem_datatypes.h"
 #include "mpi.h"
 #include "mpiimpl.h"
-#include "mpidu_shm.h"
+#include "mpidu_init_shm.h"
```
Comment (minsii) [inline src/mpid/ch3/channels/nemesis/include/mpid_nem_defs.h]: Can you please combine the three commits into a single commit? They all look like same functionality to me.

**Comment by gcongiu** [comment]
test:jenkins/ch3/tcp
test:jenkins/ch4/ofi

**Pull request merged**